### PR TITLE
Implement addTLMBus & addConnectorToTLMBus

### PR DESCRIPTION
**Pull request by robbr48**
### Related PRs

OpenModelica/OMSimulator-testsuite#83

### Purpose

- Implement addTLMBus
- Implement addConnectorToTLMBus

### Type of Change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

### Checklist

- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/OMSimulatorLib/OMSimulator.cpp [hunk 1]
```diff
@@ -372,6 +372,24 @@ oms_status_enu_t oms3_addBus(const char *cref)
...
+  oms3::ComRef tail(cref);
+  oms3::ComRef modelCref = tail.pop_front();
+  oms3::ComRef systemCref = tail.pop_front();
+  oms3::Model* model = oms3::Scope::GetInstance().getModel(modelCref);
+  if(!model) {
+    return logError("Model \"" + std::string(modelCref) + "\" does not exist in the scope");
+  }
+  oms3::System* system = model->getSystem(systemCref);
+  if(!system) {
+    return logError("Model \"" + std::string(modelCref) + "\" does not contain system \"" + std::string(systemCref) + "\"");
```
Comment (lochel) [inline src/OMSimulatorLib/OMSimulator.cpp]: Use `logError_SystemNotInModel(modelCref, systemCref)` instead.

File: src/OMSimulatorLib/OMSimulator.cpp [hunk 2]
```diff
@@ -372,6 +372,24 @@ oms_status_enu_t oms3_addBus(const char *cref)
...
+oms_status_enu_t oms3_addTLMBus(const char *cref, const char *domain, const int dimensions, const oms_tlm_interpolation_t interpolation)
+{
+  //todo Fix!
+  logTrace();
+  oms3::ComRef tail(cref);
+  oms3::ComRef modelCref = tail.pop_front();
+  oms3::ComRef systemCref = tail.pop_front();
+  oms3::Model* model = oms3::Scope::GetInstance().getModel(modelCref);
+  if(!model) {
+    return logError("Model \"" + std::string(modelCref) + "\" does not exist in the scope");
```
Comment (lochel) [inline src/OMSimulatorLib/OMSimulator.cpp]: Use `logError_ModelNotInScope(modelCref)` instead.

File: src/OMSimulatorLib/OMSimulator.cpp [hunk 3]
```diff
@@ -372,6 +372,24 @@ oms_status_enu_t oms3_addBus(const char *cref)
   return system->addBus(tail);
 }
 
+oms_status_enu_t oms3_addTLMBus(const char *cref, const char *domain, const int dimensions, const oms_tlm_interpolation_t interpolation)
+{
+  //todo Fix!
```
Comment (lochel) [inline src/OMSimulatorLib/OMSimulator.cpp]: @robbr48 What needs to be fixed here?

File: src/OMSimulatorLib/OMSimulator.h [hunk 4]
```diff
@@ -73,15 +73,15 @@ oms_status_enu_t oms3_getConnections(const char* cref, oms3_connection_t*** conn
 oms_status_enu_t oms3_addConnection(const char* crefA, const char* crefB);
 oms_status_enu_t oms3_addBus(const char* cref);
 oms_status_enu_t oms3_addConnectorToBus(const char* busCref, const char* connectorCref);
+oms_status_enu_t oms3_addTLMBus(const char* cref, const char* domain, const int dimensions, const oms_tlm_interpolation_t interpolation);
+oms_status_enu_t oms3_addConnectorToTLMBus(const char* busCref, const char* connectorCref, const char *type);
```
Comment (lochel) [inline src/OMSimulatorLib/OMSimulator.h]: What is wrong with `oms_tlm_connector_type_enu_t`?

File: src/OMSimulatorLib/OMSimulator.cpp (hunk 3 above)
Comment (robbr48) [reply src/OMSimulatorLib/OMSimulator.cpp]: Nothing actually, I just forgot to remove the comment :)

File: src/OMSimulatorLib/OMSimulator.h (hunk 4 above)
Comment (robbr48) [reply src/OMSimulatorLib/OMSimulator.h]: That enum would need to contain almost 200 entries, to cover all combinations of dimensions, causalities and interpolation methods. This feels unrealistic. I added a list of strings with accepted variable types for each combination instead.

**Comment by OpenModelica-Hudson** [comment]
The following pull requests are now being [tested](https://test.openmodelica.org/hudson/job/OMSimulator-PR/636//console). It may take a while, depending on the workload.
  - OpenModelica/OMSimulator#373
  - OpenModelica/OMSimulator-testsuite#83

The changes will be pushed if all tests run successfully.

**Comment by OpenModelica-Hudson** [comment]
The following pull requests are now being [tested](https://test.openmodelica.org/hudson/job/OMSimulator-PR/637//console). It may take a while, depending on the workload.
  - OpenModelica/OMSimulator#373
  - OpenModelica/OMSimulator-testsuite#83

The changes will be pushed if all tests run successfully.

**Comment by OpenModelica-Hudson** [comment]
All tests were successful. :bowtie:

**Pull request closed without merging**